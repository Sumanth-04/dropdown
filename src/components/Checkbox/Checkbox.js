import React, { Component } from "react";
import RcCheckbox from "rc-checkbox";
import "./Checkbox.scss";

class CheckBox extends Component {
  constructor(){
    super();
    this.state = {
      
    }
  }
  render() {
    const { children, label ,round } = this.props;
    var radius = '3px';
    if(round) {radius = '10px';}
    return (
      <span>
        <RcCheckbox className="rd-checkbox" style={{borderRadius : radius}} {...this.props} />{" "}
        {label ? label : children}
      </span>
    );
  }
}

export default CheckBox;
