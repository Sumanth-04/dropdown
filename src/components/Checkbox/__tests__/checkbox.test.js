import React from "react";
import renderer from 'react-test-renderer';
import Checkbox from "../Checkbox";
//import {isTSAnyKeyword} from '@babel/types';

it("Checkbox should have a label", () => {
    const tree = renderer
      .create(
        <Checkbox label="Check me!!"></Checkbox>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
});