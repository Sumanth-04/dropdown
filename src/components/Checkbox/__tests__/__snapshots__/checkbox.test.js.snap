// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should have a label 1`] = `
<span>
  <span
    className="rc-checkbox rd-checkbox"
    style={
      Object {
        "borderRadius": "3px",
      }
    }
  >
    <input
      checked={false}
      className="rc-checkbox-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <span
      className="rc-checkbox-inner"
    />
  </span>
   
  Check me!!
</span>
`;

exports[`Rendering Checkbox 1`] = `
<span>
  <span
    className="rc-checkbox rd-checkbox"
    style={
      Object {
        "borderRadius": "3px",
      }
    }
  >
    <input
      checked={false}
      className="rc-checkbox-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <span
      className="rc-checkbox-inner"
    />
  </span>
   
</span>
`;

exports[`Rendering Checkbox 2`] = `
<span>
  <span
    className="rc-checkbox rd-checkbox"
    style={
      Object {
        "height": "20px",
        "width": "20px",
      }
    }
  >
    <input
      checked={false}
      className="rc-checkbox-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <span
      className="rc-checkbox-inner"
    />
  </span>
   
</span>
`;
